src/controller/artifact/processor/default.go

```diff
@@ -44,16 +55,113 @@ func (d *defaultProcessor) GetArtifactType() string {
+		return nil
+	}
+	// get config layer
+	_, blob, err := d.RegCli.PullBlob(artifact.RepositoryName, mani.Config.Digest.String())
+	if err != nil {
+		log.Warningf("get config layer error: %v", err)
+		return nil
+	}
+	defer blob.Close()
+	// parse metadata from config layer
+	metadata := map[string]interface{}{}
+	if err := json.NewDecoder(blob).Decode(&metadata); err != nil {
+		log.Warningf("parse config layer error: %v", err)
+		return nil
+	}
+	// parse schema
+	schemaMap, ok := metadata[schemaV1.XHarborAttributes]
+	if !ok {
+		return nil
+	}
+	schema := &schemaV1.Schema{}
+	err = mapstructure.Decode(schemaMap, schema)
+	if err != nil {
+		log.Warningf("unsupported artifact config schema: %v", err)
+		return nil
+	}
+	// check schema version
+	if schema.SchemaVersion != schemaV1.SchemaVersionV1 {
+		log.Warningf("unsupported artifact config schema version %d", schema.SchemaVersion)
+		return nil
+	}
+	// check icon url
+	_, err = url.ParseRequestURI(schema.Icon)
+	if err != nil {
+		log.Warningf("icon url invalid %v", err)
+		schema.Icon = ""
+		m := make(map[string]interface{})
+		b, _ := json.Marshal(schema)
+		_ = json.Unmarshal(b, &m)
+		metadata[schemaV1.XHarborAttributes] = m
+	}
+	// move keys in skipKeyList
+	if schema.SkipKeyList != nil {
+		for _, denyKey := range schema.SkipKeyList {
+			delete(metadata, denyKey)
+		}
+	}
+	artifact.ExtraAttrs = metadata
+
 	return nil
 }
 func (d *defaultProcessor) AbstractAddition(ctx context.Context, artifact *artifact.Artifact, addition string) (*Addition, error) {
-	// return error directly
-	return nil, errors.New(nil).WithCode(errors.BadRequestCode).
-		WithMessage("the processor for artifact %s not found, cannot get the addition", artifact.Type)
+	schemaMap, ok := artifact.ExtraAttrs[schemaV1.XHarborAttributes]
+	if !ok {
+		return nil, errors.New(nil).WithCode(errors.BadRequestCode).
+			WithMessage("the processor for artifact %s not found, cannot get the addition", artifact.Type)
+	}
+	schema := &schemaV1.Schema{}
+	mapstructure.Decode(schemaMap, schema)
+	additionSchemas := make(map[string]schemaV1.AdditionSchema)
+	for _, addition := range schema.Additions {
+		additionSchemas[addition.Name] = addition
+	}
+	if _, ok := additionSchemas[addition]; !ok {
+		return nil, errors.New(nil).WithCode(errors.BadRequestCode).
+			WithMessage("addition %s isn't supported for %s", addition, artifact.Type)
+	}
+	contentType := additionSchemas[addition].ContentType
+	if contentType != "text/plain" {
+		return nil, errors.New(nil).WithCode(errors.BadRequestCode).
+			WithMessage("unsupported content type %s in addition", contentType)
+	}
+	layerDigest := additionSchemas[addition].Digest
+	_, blob, err := d.RegCli.PullBlob(artifact.RepositoryName, layerDigest)
+	if err != nil {
+		return nil, err
+	}
+	content, err := ioutil.ReadAll(blob)
+	if err != nil {
+		return nil, err
+	}
+	blob.Close()
+
+	return &Addition{
+		Content:     content,
+		ContentType: "text/plain; charset=utf-8",
```

Comment:
Use the content type read from the config

Reply:
ok